code/modules/clothing/modular_armor/tdf.dm

```diff
@@ -0,0 +1,363 @@
+		/obj/item/armor_module/storage/medical,
+	)
+
+//helmet
+
+/obj/item/clothing/head/modular/tdf
+	name = "\improper Pawn Class Helmet"
+	desc = "The standard combat helmet worn by TDF combat troops. Comes with an integrated hud and AR to provide situational awareness to the wearer."
+	icon = 'icons/mob/modular/tdf_helmets.dmi'
+	worn_icon_list = list(
+		slot_head_str = 'icons/mob/modular/tdf_helmets.dmi',
+		slot_l_hand_str = 'icons/mob/inhands/clothing/hats_left.dmi',
+		slot_r_hand_str = 'icons/mob/inhands/clothing/hats_right.dmi',
+	)
+	icon_state = "tdf_helmet"
+	worn_icon_state = "tdf_helmet"
+	soft_armor = MARINE_ARMOR_HEAVY
+	inv_hide_flags = HIDEEARS|HIDEALLHAIR
+	armor_protection_flags = HEAD|FACE|EYES
+	attachments_allowed = list(
+		/obj/item/armor_module/module/tyr_head,
+		/obj/item/armor_module/module/fire_proof_helmet,
+		/obj/item/armor_module/module/hod_head,
+		/obj/item/armor_module/module/mimir_environment_protection/mimir_helmet,
+		/obj/item/armor_module/module/mimir_environment_protection/mimir_helmet/mark1,
+		/obj/item/armor_module/module/welding,
+		/obj/item/armor_module/module/welding/superior,
+		/obj/item/armor_module/module/binoculars,
+		/obj/item/armor_module/module/binoculars/artemis_mark_two,
+		/obj/item/armor_module/module/artemis,
+		/obj/item/armor_module/module/antenna,
+		/obj/item/armor_module/module/night_vision,
+		/obj/item/armor_module/storage/helmet,
+		/obj/item/armor_module/armor/badge,
+	)
+	greyscale_config = null
+	starting_attachments = list(/obj/item/armor_module/storage/helmet)
+	icon_state_variants = list(
+		"black",
+	)
+	current_variant = "black"
+	visorless_offset_y = 0
+
+/obj/item/clothing/head/modular/tdf/hod
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/hod_head)
+
+/obj/item/clothing/head/modular/tdf/freyr
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/artemis)
+
+/obj/item/clothing/head/modular/tdf/antenna
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/antenna)
+
+/obj/item/clothing/head/modular/tdf/welding
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/welding)
+
+/obj/item/clothing/head/modular/tdf/superiorwelding
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/welding/superior)
+
+/obj/item/clothing/head/modular/tdf/mimir
+	starting_attachments = list(/obj/item/armor_module/storage/helmet, /obj/item/armor_module/module/mimir_environment_protection/mimir_helmet/mark1)
+
+/obj/item/clothing/head/modular/tdf/medium
+	name = "\improper Knight Class Helmet"
+	icon_state = "tdf_helmet_medium"
+	worn_icon_state = "tdf_helmet_medium"
+	attachments_allowed = list(
+		/obj/item/armor_module/storage/helmet,
+		/obj/item/armor_module/armor/badge,
+	)
+
+/obj/item/clothing/head/modular/tdf/heavy
```

Comment:
Up to you, but since you've made the effort to make different helmets for the different armour classes, make their armour values vary appropriately would be good as well. (SOM actually do this, although people don't realise, mainly due to lack of sprites. Standard SOM had has medium armour, veteran SOM hat has heavy armour)